core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -221,6 +224,10 @@ private boolean prepareResponse(final ClusterHealthRequest request, final Cluste
         if (request.waitForNoRelocatingShards() && response.getRelocatingShards() == 0) {
             waitForCounter++;
         }
+        if (request.waitForNoInitializingShards() && !clusterState.getRoutingNodes().hasInactivePrimaries() &&
```

Comment:
you can use the same approach as for the `waitForNoRelocatingShards` a few lines above, i.e. `request.waitForNoInitializingShards() && response.getInitializingShards() == 0`